hpx/util/memory_chunk_pool.hpp

```diff
@@ -106,19 +117,99 @@ namespace hpx { namespace util {
                 }
             }
             {
-                mutex_type::scoped_lock l(large_chunks_mtx_);
+                typename mutex_type::scoped_lock l(large_chunks_mtx_);
                 large_chunks_.insert(std::make_pair(size, p));
             }
         }
 
         mutable mutex_type chunks_mtx_;
-        std::vector<memory_chunk> memory_chunks_;
+        std::vector<memory_chunk_type> memory_chunks_;
         const std::size_t chunk_size_;
         const std::size_t max_chunks_;
 
         mutable mutex_type large_chunks_mtx_;
         large_chunks_type large_chunks_;
     };
+
+    namespace detail
+    {
+        template <typename Mutex>
+        struct memory_chunk_pool_allocator
+        {
+            typedef std::size_t size_type;
+            typedef std::ptrdiff_t difference_type;
+            typedef char* pointer;
+            typedef const char* const_pointer;
+            typedef char& reference;
+            typedef const char& const_reference;
+            typedef char value_type;
+
+            template <typename U>
+            struct rebind
+            {
+                typedef memory_chunk_pool_allocator other;
+            };
```

Comment:
This implementation of rebind is not conforming. it needs to return a new allocator type which allocates objects of type `U`. This rebind always returns an allocator which allocates `char`. In conforming implementations this will not compile.